tracing-attributes/tests/follows_from.rs

```diff
@@ -99,3 +115,28 @@ fn follows_from_current_test() {
 
     handle.assert_finished();
 }
+
+#[test]
+fn follows_from_noncopy_test() {
+    let cause_a = expect::span().named("cause_a");
+    let consequence = expect::span().named("follows_from_noncopy");
+
+    let (collector, handle) = collector::mock()
+        .new_span(cause_a.clone())
+        .new_span(consequence.clone())
+        .follows_from(consequence.clone(), cause_a)
+        .enter(consequence.clone())
+        .exit(consequence)
+        .only()
+        .run_with_handle();
+
+    with_default(collector, || {
+        let context = NonCopyContext {
+            id: Span::current().id(),
+        };
+
+        follows_from_noncopy(context)
+    });
+
+    handle.assert_finished();
+}
```

Comment:
hmm, looks like this test is failing because we never actually create the span `cause_a`: https://github.com/tokio-rs/tracing/actions/runs/6017840647/job/16324905675?pr=2686#step:5:973

Reply:
I am not sure why we're not creating the span. On Line 121 and 125 the "cause_a" span isused... maybe I am missing some background information on how the tests work? :thinking:

Reply:
on line 121, you're saying that you want to _expect_ a span named `cause_a`, so the subscriber knows to assert that the span is created and entered. but your test never actually _creates_ that span, so the assertion fails, because it never sees that span being created.

i think the test will need to be changed to look like this:
```suggestion
fn follows_from_noncopy_test() {
    let cause_a = expect::span().named("cause_a");
    let consequence = expect::span().named("follows_from_noncopy");

    let (collector, handle) = collector::mock()
        .new_span(cause_a.clone())
        .new_span(consequence.clone())
        .follows_from(consequence.clone(), cause_a)
        .enter(consequence.clone())
        .exit(consequence)
        .only()
        .run_with_handle();

    with_default(collector, || {
        let cause_a = tracing::info_span!("cause_a");
        let context = NonCopyContext {
            id:  cause_a.id().unwrap(),
        };

        follows_from_noncopy(context)
    });

    handle.assert_finished();
}
```